### PR TITLE
Add support for creating and restoring data set

### PR DESCRIPTION
**Pull request by darinkrauss**
- Logout when service is deleted
- Add bundle semantic version

### CODE REVIEW

File: TidepoolServiceKit/TidepoolService.swift [hunk 1]
```diff
@@ -45,25 +54,74 @@ public final class TidepoolService: Service {
...
+        guard let session = session, let clientName = Bundle.main.bundleIdentifier else {
+            return
+        }
+        tapi.listDataSets(filter: TDataSet.Filter(clientName: clientName), session: session) { result in
+            switch result {
+            case .failure(let error):
+                self.error = error
+            case .success(let dataSets):
+                if !dataSets.isEmpty {
+                    self.dataSetId = dataSets.first?.uploadId
```
Comment (rickpasetto) [inline TidepoolServiceKit/TidepoolService.swift:108]: Sorry I'm not familiar with the Tidepool API, but is it customary to only grab the first `dataSet`?  What if there are more?

File: TidepoolServiceKit/TidepoolService.swift [hunk 2]
```diff
@@ -45,25 +54,74 @@ public final class TidepoolService: Service {
...
 
     public func completeDelete() {
+        guard let session = session else {
+            return
+        }
+
         self.session = nil
+
+        DispatchQueue.global(qos: .background).async {
+            self.tapi.logout(session: session) { _ in }
```
Comment (rickpasetto) [inline TidepoolServiceKit/TidepoolService.swift:91]: Sorry, again, I'm not sure I'm understanding the flow here, but, why does `completeDelete` logout?  Is this method intended to delete the data set or delete the "service"?

**Review by rickpasetto — COMMENTED** [review]
Looks ok so far (would love to see some unit tests), but I had some questions, if you have a chance...

File: TidepoolServiceKit/TidepoolService.swift (hunk 1 above)
Comment (darinkrauss) [reply TidepoolServiceKit/TidepoolService.swift:108]: This is searching for a data set that is specific to Tidepool Loop. There should only be one of them at any time. So, grabbing the first one is appropriate here. However, I will add a check and log if there is more than one.

File: TidepoolServiceKit/TidepoolService.swift (hunk 2 above)
Comment (darinkrauss) [reply TidepoolServiceKit/TidepoolService.swift:91]: `completeDelete` is invoked when the service is being deleted to do any final clean before all of the service's settings are destroyed. A `logout` is appropriate in this case.

**Comment by darinkrauss** [comment]
@rickpasetto The latest commit adds logging if there was more than one data set found. Would you please review?

@ps2 I added you since @nhamming is likely out for a while.

File: TidepoolServiceKit/TidepoolService.swift (hunk 1 above)
Comment (rickpasetto) [reply TidepoolServiceKit/TidepoolService.swift:108]: Sure, or a comment is fine.  Thanks.

File: TidepoolServiceKit/TidepoolService.swift (hunk 2 above)
Comment (rickpasetto) [reply TidepoolServiceKit/TidepoolService.swift:91]: Gotcha.

File: TidepoolServiceKit/TidepoolService.swift [hunk 3]
```diff
@@ -99,6 +102,9 @@ public final class TidepoolService: Service {
                 self.error = error
             case .success(let dataSets):
                 if !dataSets.isEmpty {
+                    if dataSets.count > 1 {
+                        self.log.default("Found multiple matching data sets; expected zero or one")
```
Comment (rickpasetto) [inline TidepoolServiceKit/TidepoolService.swift]: definitely a [nit] : I would've expected this to be `log.error`...but this is fine as-is.

**Review by rickpasetto — DISMISSED** [review]
One nit,  otherwise LGTM.

**Comment by darinkrauss** [comment]
@rickpasetto I changed the log `default` to `error` per your suggestion. Also, I accidentally re-requested your review, so if you could approve again that'd be great. Thanks!

**Review by rickpasetto — APPROVED** [review]
LGTM!

**Review by ps2 — APPROVED** [review]
LGTM!

**Pull request merged**